app/services/umrdr/date_range_service.rb

```diff
@@ -0,0 +1,78 @@
+module Umrdr
+  class DateRangeService  
+
+    attr_reader :params
+    
+    def initialize(params)
+      @params = params
+    end  
+
+    def get_yr_mon_day(date_param)
+      if params[date_param].length>0 && params[date_param].match(/^\d+$/)
+        val = params[date_param].to_i
+      else
+        val = nil 
+      end
+      return val
+    end  
+
+    def get_precision_date(y,m,d)
+      args = []
+
+      if y 
+        args << y
+        precision = :"year_precision!"
+      end
+
+      if m 
+        args << m
+        precision = :"month_precision!"
+      end
+
+      if d
+        args << d
+        precision = :"day_precision!"
+      end
+      
+      if y 
+        val = Date.new(*args)
+        val.send(precision)
+        val = val.edtf
+      else
+        val = nil
+      end
+        
+      return val
+    end  
+
+    def transform
+
+      y = get_yr_mon_day(:date_coverage_1_year)
+      m = get_yr_mon_day(:date_coverage_1_month)
+      d = get_yr_mon_day(:date_coverage_1_day)
+
+      date_from = get_precision_date(y,m,d)
+      
+      y = get_yr_mon_day(:date_coverage_2_year)
+      m = get_yr_mon_day(:date_coverage_2_month)
+      d = get_yr_mon_day(:date_coverage_2_day)
+
+      date_to = get_precision_date(y,m,d)
+      
+    
+      if (date_from && date_to)    
+        dateC = date_from + '/' + date_to
+        date_interval = Date.edtf(dateC)
+      elsif (date_from)
+        date_interval = date_from
+      elsif (date_to)
+        date_interval = '/'+ date_to
+      else
+        date_interval = ""
+      end    
+
+    return date_interval.to_s
```

Comment:
This returns empty string in the case the date could not be parsed, or was not entered.  That will lead to the date_coverage property being `[""]` instead of remaining `[]`.